### PR TITLE
Cookie refresh

### PR DESCRIPTION
**Pull request by markjamesbutler**
This PR creates an a/b test to verify the fix for the [persistent sign in for existing users](https://trello.com/c/kszkBW8N/225-2-persistent-sign-in-for-existing-users) bug.

We noticed that since the move to next-gen, active users are no longer having their sessions extended.  This fix implements similar logic to the implementation in R2.  Essentially, it refreshes user cookies.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/tests/cookie-refresh.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        this.expiry = '2015-06-14';
+        this.author = 'Mark Butler';
+        this.description = 'Measured roll out of Identity session refresh';
+        this.audience = 0.01;
+        this.audienceOffset = 0;
+        this.successMeasure = '';
+        this.audienceCriteria = '1% of users.';
+        this.dataLinkNames = '';
+        this.idealOutcome = '1% of users have their signed in session extended. If they are signed-in and their session is over one month old.';
+        this.showForSensitive = false;
```
Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/experiments/tests/cookie-refresh.js]: is it really a problem for sensitive articles?

File: static/src/javascripts/projects/common/modules/identity/api.js [hunk 2]
```diff
@@ -185,6 +186,21 @@ define([
         return false;
     };
 
+    Id.refreshCookie = function () {
+        if (Id.isUserLoggedIn()) {
+            var lastRefresh = storage.local.get(Id.lastRefreshKey),
```
Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/identity/api.js]: if this is not set in local storage, will this code ever refresh the cookie?

**Comment by johnduffell** [comment]
:+1: interesting use of AB test framework but it's all good

File: static/src/javascripts/projects/common/modules/identity/api.js (hunk 2 above)
Comment (markjamesbutler) [inline static/src/javascripts/projects/common/modules/identity/api.js]: Yes see 'should return true for a user who has never refreshed cookies' in Api.spec

Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/identity/api.js]: unit testing is awesome! =)

**Pull request merged**